pilot/pkg/xds/waypoint_test.go

```diff
@@ -218,6 +218,81 @@ func TestWaypoint(t *testing.T) {
 	})
 }
 
+func TestWaypointTLSInspector(t *testing.T) {
```

Comment:
we might need 1 test for both inspectors, given the overlapping behaviors for them. I worry the change (discussed in another comment) might have some impact on the existing HTTP inspector code?

Reply:
Done